File: scripts/test-student-chat.py
#!/usr/bin/env python3
"""
学生移动端 AI 对话功能测试脚本
"""
import asyncio
from playwright.async_api import async_playwright
import os

# 配置
BASE_URL = "http://localhost:5174"
CHAT_URL = f"{BASE_URL}/pages/chat/index"
LOGIN_URL = f"{BASE_URL}/pages/login/index"
TEST_ACCOUNT = "4523570155"
TEST_PASSWORD = "admin123"

# 截图保存目录
SCREENSHOT_DIR = "docs/test-reports/student-chat-test"
os.makedirs(SCREENSHOT_DIR, exist_ok=True)


async def test_student_chat():
    """测试学生移动端 AI 对话功能"""
    async with async_playwright() as p:
        # 启动浏览器
        browser = await p.chromium.launch(headless=True)
        context = await browser.new_context(viewport={'width': 375, 'height': 812})
        page = await context.new_page()
        
        print("=== 学生移动端 AI 对话功能测试 ===\n")
        
        # ========== 步骤 1: 访问登录页面 ==========
        print("步骤 1: 访问登录页面")
        await page.goto(LOGIN_URL)
        await page.wait_for_load_state('networkidle')
        await page.screenshot(path=f"{SCREENSHOT_DIR}/01-login-page.png")
        print("✓ 登录页面截图已保存")
        
        # ========== 步骤 2: 填写登录信息 ==========
        print("\n步骤 2: 填写登录信息")
        
        # 填写学号
        await page.fill('input[placeholder="请输入学号"]', TEST_ACCOUNT)
        print(f"✓ 填写学号: {TEST_ACCOUNT}")
        
        # 填写密码
        await page.fill('input[placeholder="请输入密码"]', TEST_PASSWORD)
        print(f"✓ 填写密码: {'*' * len(TEST_PASSWORD)}")
        
        # 获取验证码答案
        captcha_img = await page.locator('.captcha-img').get_attribute('src')
        print(f"✓ 验证码图片已加载")
        
        # 等待用户查看验证码（这里需要手动输入或OCR识别）
        # 由于验证码是数学表达式，我们先截图查看
        await page.screenshot(path=f"{SCREENSHOT_DIR}/02-form-filled.png")
        print("✓ 表单填写截图已保存")
        
        # 获取验证码图片并查看
        captcha_element = page.locator('.captcha-img')
        await captcha_element.screenshot(path=f"{SCREENSHOT_DIR}/captcha-image.png")
        print("✓ 验证码图片已保存")
        
        # 由于验证码需要识别，这里先暂停并显示提示
        print("\n注意: 需要手动识别验证码")
        print(f"验证码图片保存在: {SCREENSHOT_DIR}/captcha-image.png")
        
        # 尝试自动识别简单数学验证码 (格式通常是 "a+b=?" 或 "a*b=?")
        # 这里我们先读取验证码图片的 alt 或 title 属性（如果有的话）
        # 或者我们可以通过其他方式获取验证码答案
        
        # 暂时暂停，等待手动输入验证码
        # 在实际自动化中，可以集成 OCR 服务
        
        print("\n=== 登录步骤需要验证码识别，测试暂停 ===")
        print("请查看验证码图片并手动登录，然后运行后续测试")
        
        await browser.close()
        return False


async def test_chat_functionality(context):
    """测试聊天功能（需要已登录的 context）"""
    page = await context.new_page()
    
    print("\n=== 测试 AI 对话功能 ===\n")
    
    # ========== 进入聊天页面 ==========
    print("步骤 1: 进入聊天页面")
    await page.goto(CHAT_URL)
    await page.wait_for_load_state('networkidle')
    await page.screenshot(path=f"{SCREENSHOT_DIR}/05-chat-initial.png")
    print("✓ 聊天页面初始状态截图已保存")
    
    # 检查页面元素
    title = await page.locator('.title').text_content()
    print(f"✓ 页面标题: {title}")
    
    # 检查空状态
    empty_state = await page.locator('.empty-state').is_visible()
    if empty_state:
        print("✓ 空状态显示正常")
        empty_title = await page.locator('.empty-title').text_content()
        print(f"  - 空状态标题: {empty_title}")
    
    # 检查快捷问题
    quick_chips = await page.locator('.chip').count()
    print(f"✓ 快捷问题数量: {quick_chips}")
    
    # 检查输入框
    input_visible = await page.locator('.message-input').is_visible()
    print(f"✓ 输入框显示: {input_visible}")
    
    # ========== 测试快捷问题 ==========
    print("\n步骤 2: 测试快捷问题")
    chip = page.locator('.chip').first
    chip_text = await chip.text_content()
    print(f"✓ 点击快捷问题: {chip_text}")
    await chip.click()
    await page.wait_for_timeout(1000)
    await page.screenshot(path=f"{SCREENSHOT_DIR}/06-quick-question-clicked.png")
    print("✓ 快捷问题点击后截图已保存")
    
    # 等待 AI 响应
    print("\n步骤 3: 等待 AI 响应")
    try:
        # 等待消息出现
        await page.wait_for_selector('.message-item.user', timeout=5000)
        print("✓ 用户消息已显示")
        
        # 等待 AI 响应
        await page.wait_for_selector('.message-item.assistant', timeout=30000)
        print("✓ AI 响应已显示")
        
        # 等待流式输出完成
        await page.wait_for_timeout(5000)
        await page.screenshot(path=f"{SCREENSHOT_DIR}/07-ai-response.png")
        print("✓ AI 响应截图已保存")
        
        # 检查 AI 消息内容
        ai_message = await page.locator('.message-item.assistant .message-text').text_content()
        print(f"✓ AI 回复内容: {ai_message[:100]}...")
        
    except Exception as e:
        print(f"✗ AI 响应测试失败: {e}")
        await page.screenshot(path=f"{SCREENSHOT_DIR}/07-ai-response-error.png")
    
    # ========== 测试手动输入 ==========
    print("\n步骤 4: 测试手动输入")
    test_message = "奖学金怎么申请？"
    await page.fill('.message-input', test_message)
    await page.click('.send-btn')
    print(f"✓ 发送消息: {test_message}")
    
    await page.wait_for_timeout(1000)
    await page.screenshot(path=f"{SCREENSHOT_DIR}/08-manual-message-sent.png")
    print("✓ 手动消息发送后截图已保存")
    
    # 等待 AI 响应
    try:
        await page.wait_for_timeout(10000)
        await page.screenshot(path=f"{SCREENSHOT_DIR}/09-manual-response.png")
        print("✓ 手动消息 AI 响应截图已保存")
    except Exception as e:
        print(f"✗ 手动消息测试失败: {e}")
    
    await page.close()


if __name__ == "__main__":
    asyncio.run(test_student_chat())

File: scripts/test-student-chat.js
#!/usr/bin/env node
/**
 * 学生移动端 AI 对话功能测试脚本 (Node.js + Playwright)
 */
const { chromium } = require('playwright');
const path = require('path');
const fs = require('fs');

// 配置
const BASE_URL = 'http://localhost:5174';
const CHAT_URL = `${BASE_URL}/pages/chat/index`;
const LOGIN_URL = `${BASE_URL}/pages/login/index`;
const TEST_ACCOUNT = '4523570155';
const TEST_PASSWORD = 'admin123';

// 截图保存目录
const SCREENSHOT_DIR = path.join(__dirname, '..', 'docs', 'test-reports', 'student-chat-test');
if (!fs.existsSync(SCREENSHOT_DIR)) {
  fs.mkdirSync(SCREENSHOT_DIR, { recursive: true });
}

async function delay(ms) {
  return new Promise(resolve => setTimeout(resolve, ms));
}

async function testStudentChat() {
  console.log('=== 学生移动端 AI 对话功能测试 ===\n');
  
  // 启动浏览器
  const browser = await chromium.launch({ headless: true });
  const context = await browser.newContext({
    viewport: { width: 375, height: 812 }
  });
  const page = await context.newPage();
  
  try {
    // ========== 步骤 1: 访问登录页面 ==========
    console.log('步骤 1: 访问登录页面');
    await page.goto(LOGIN_URL, { waitUntil: 'networkidle' });
    await page.screenshot({ path: path.join(SCREENSHOT_DIR, '01-login-page.png') });
    console.log('✓ 登录页面截图已保存');
    
    // ========== 步骤 2: 填写登录信息 ==========
    console.log('\n步骤 2: 填写登录信息');
    
    // 等待页面完全加载
    await page.waitForLoadState('domcontentloaded');
    await delay(1000);
    
    // 使用更通用的选择器
    const inputs = await page.locator('input').all();
    console.log(`找到 ${inputs.length} 个输入框`);
    
    // 填写学号（第一个输入框）
    if (inputs.length > 0) {
      await inputs[0].fill(TEST_ACCOUNT);
      console.log(`✓ 填写学号: ${TEST_ACCOUNT}`);
    }
    
    // 填写密码（第二个输入框）
    if (inputs.length > 1) {
      await inputs[1].fill(TEST_PASSWORD);
      console.log(`✓ 填写密码: ${'*'.repeat(TEST_PASSWORD.length)}`);
    }
    
    await page.screenshot({ path: path.join(SCREENSHOT_DIR, '02-form-filled.png') });
    console.log('✓ 表单填写截图已保存');
    
    // 保存验证码图片
    const captchaElement = page.locator('.captcha-img');
    await captchaElement.screenshot({ path: path.join(SCREENSHOT_DIR, 'captcha-image.png') });
    console.log('✓ 验证码图片已保存');
    
    console.log('\n步骤 3: 识别并填写验证码');
    
    // 读取验证码图片并识别
    // 当前验证码是数学表达式 "5+2=?"
    // 我们需要通过 OCR 或手动识别
    // 这里为了测试，我们先读取当前显示的验证码
    
    // 由于验证码是动态生成的，我们需要等待验证码加载完成
    await delay(500);
    
    // 重新获取验证码截图
    await captchaElement.screenshot({ path: path.join(SCREENSHOT_DIR, 'captcha-current.png') });
    
    // 注意：这里需要 OCR 识别验证码
    // 由于当前验证码是 "5+2=?", 答案是 7
    // 实际测试时需要根据当前显示的验证码填写正确答案
    
    // 先尝试使用固定值 7 (5+2=7)，但需要注意验证码可能已经刷新
    // 让我们先查看当前的验证码是什么
    
    // 从页面获取验证码图片的 src
    const captchaSrc = await captchaElement.getAttribute('src');
    if (captchaSrc) {
      console.log('✓ 验证码图片已加载');
    }
    
    // 由于无法自动识别验证码，我们尝试使用后端 API 直接登录
    // 或者跳过登录测试，直接基于代码分析进行测试报告
    
    console.log('\n注意：验证码需要手动识别');
    console.log(`当前验证码图片: ${path.join(SCREENSHOT_DIR, 'captcha-current.png')}`);
    console.log('由于验证码是动态生成的，自动化测试受限');
    
    // 尝试填写验证码并登录
    // 注意：这里填写的是之前截图中的验证码答案，可能已经过期
    const captchaAnswer = '7'; // 假设当前验证码仍是 5+2=?
    if (inputs.length > 2) {
      await inputs[2].fill(captchaAnswer);
      console.log(`✓ 尝试填写验证码: ${captchaAnswer}`);
    }
    
    await page.screenshot({ path: path.join(SCREENSHOT_DIR, '03-captcha-filled.png') });
    console.log('✓ 验证码填写截图已保存');
    
    // 点击登录按钮
    console.log('\n步骤 4: 点击登录');
    await page.click('.login-btn');
    await delay(3000);
    
    // 检查是否登录成功
    const currentUrl = page.url();
    console.log(`当前页面: ${currentUrl}`);
    
    if (currentUrl.includes('/home') || currentUrl.includes('/chat')) {
      console.log('✓ 登录成功！');
      
      // 继续测试聊天功能
      await testChatFunctionality(page);
    } else {
      console.log('✗ 登录失败，仍在登录页面');
      await page.screenshot({ path: path.join(SCREENSHOT_DIR, '03-login-failed.png') });
      
      // 检查是否有错误提示
      const toastVisible = await page.locator('.uni-toast').isVisible().catch(() => false);
      if (toastVisible) {
        const toastText = await page.locator('.uni-toast').textContent();
        console.log(`错误提示: ${toastText}`);
      }
    }
    
  } catch (error) {
    console.error('测试出错:', error);
    await page.screenshot({ path: path.join(SCREENSHOT_DIR, 'error-screenshot.png') });
  } finally {
    await browser.close();
  }
}

async function testChatFunctionality(page) {
  console.log('\n=== 测试 AI 对话功能 ===\n');
  
  // ========== 进入聊天页面 ==========
  console.log('步骤 1: 进入聊天页面');
  await page.goto(CHAT_URL, { waitUntil: 'networkidle' });
  await page.screenshot({ path: path.join(SCREENSHOT_DIR, '05-chat-initial.png') });
  console.log('✓ 聊天页面初始状态截图已保存');
  
  // 检查页面元素
  const title = await page.locator('.title').textContent();
  console.log(`✓ 页面标题: ${title}`);
  
  // 检查空状态
  const emptyStateVisible = await page.locator('.empty-state').isVisible().catch(() => false);
  if (emptyStateVisible) {
    console.log('✓ 空状态显示正常');
    const emptyTitle = await page.locator('.empty-title').textContent();
    console.log(`  - 空状态标题: ${emptyTitle}`);
    
    const emptyDesc = await page.locator('.empty-desc').textContent();
    console.log(`  - 空状态描述: ${emptyDesc}`);
  }
  
  // 检查快捷问题
  const quickChips = await page.locator('.chip').count();
  console.log(`✓ 快捷问题数量: ${quickChips}`);
  
  // 显示快捷问题内容
  if (quickChips > 0) {
    const chipTexts = await page.locator('.chip').allTextContents();
    console.log(`  - 快捷问题: ${chipTexts.join(', ')}`);
  }
  
  // 检查输入框
  const inputVisible = await page.locator('.message-input').isVisible().catch(() => false);
  console.log(`✓ 输入框显示: ${inputVisible}`);
  
  // ========== 测试快捷问题 ==========
  if (quickChips > 0) {
    console.log('\n步骤 2: 测试快捷问题');
    const chip = page.locator('.chip').first;
    const chipText = await chip.textContent();
    console.log(`✓ 点击快捷问题: ${chipText}`);
    await chip.click();
    await delay(1000);
    await page.screenshot({ path: path.join(SCREENSHOT_DIR, '06-quick-question-clicked.png') });
    console.log('✓ 快捷问题点击后截图已保存');
    
    // 等待 AI 响应
    console.log('\n步骤 3: 等待 AI 响应');
    try {
      // 等待用户消息出现
      await page.waitForSelector('.message-item.user', { timeout: 5000 });
      console.log('✓ 用户消息已显示');
      
      // 等待 AI 响应（最多30秒）
      await page.waitForSelector('.message-item.assistant', { timeout: 30000 });
      console.log('✓ AI 响应已显示');
      
      // 等待流式输出完成
      await delay(5000);
      await page.screenshot({ path: path.join(SCREENSHOT_DIR, '07-ai-response.png') });
      console.log('✓ AI 响应截图已保存');
      
      // 检查 AI 消息内容
      const aiMessage = await page.locator('.message-item.assistant .message-text').textContent();
      console.log(`✓ AI 回复内容预览: ${aiMessage.substring(0, 100)}...`);
      
      // 检查是否有来源引用
      const sourcesVisible = await page.locator('.message-sources').isVisible().catch(() => false);
      console.log(`✓ 来源引用显示: ${sourcesVisible}`);
      
    } catch (e) {
      console.log(`✗ AI 响应测试失败: ${e.message}`);
      await page.screenshot({ path: path.join(SCREENSHOT_DIR, '07-ai-response-error.png') });
    }
  }
  
  // ========== 测试手动输入 ==========
  console.log('\n步骤 4: 测试手动输入');
  const testMessage = '奖学金怎么申请？';
  await page.fill('.message-input', testMessage);
  
  // 检查发送按钮是否激活
  const sendBtnActive = await page.locator('.send-btn.active').isVisible().catch(() => false);
  console.log(`✓ 发送按钮激活: ${sendBtnActive}`);
  
  await page.click('.send-btn');
  console.log(`✓ 发送消息: ${testMessage}`);
  
  await delay(1000);
  await page.screenshot({ path: path.join(SCREENSHOT_DIR, '08-manual-message-sent.png') });
  console.log('✓ 手动消息发送后截图已保存');
  
  // 等待 AI 响应
  try {
    await page.waitForSelector('.message-item.assistant', { timeout: 30000 });
    await delay(5000);
    await page.screenshot({ path: path.join(SCREENSHOT_DIR, '09-manual-response.png') });
    console.log('✓ 手动消息 AI 响应截图已保存');
    
    // 获取所有消息
    const allMessages = await page.locator('.message-item').count();
    console.log(`✓ 总消息数量: ${allMessages}`);
    
  } catch (e) {
    console.log(`✗ 手动消息测试失败: ${e.message}`);
  }
  
  // ========== 测试复制功能 ==========
  console.log('\n步骤 5: 测试复制功能');
  try {
    const copyBtn = page.locator('.copy-btn').first;
    const copyBtnVisible = await copyBtn.isVisible().catch(() => false);
    if (copyBtnVisible) {
      await copyBtn.click();
      console.log('✓ 点击复制按钮');
      await delay(500);
      
      // 检查是否有复制成功提示
      const checkIconVisible = await page.locator('.copy-btn .icon-check, .copy-btn svg[data-icon="check"]').isVisible().catch(() => false);
      console.log(`✓ 复制成功图标显示: ${checkIconVisible}`);
    } else {
      console.log('✗ 复制按钮未找到');
    }
  } catch (e) {
    console.log(`✗ 复制功能测试失败: ${e.message}`);
  }
  
  console.log('\n=== AI 对话功能测试完成 ===');
}

// 运行测试
testStudentChat().catch(console.error);
